sympy/polys/series/tests/test_powerseriesring.py

```diff
@@ -0,0 +1,486 @@
+from sympy.polys.domains import ZZ, QQ
+from sympy.polys.series.powerseriesring import PowerSeriesRing
+from sympy.polys.series.python_powerseriesring import (
+    PythonPowerSeriesRingZZ,
+    PythonPowerSeriesRingQQ,
+)
+from sympy.external.gmpy import GROUND_TYPES
+import pytest
+from sympy.testing.pytest import raises
+from sympy.polys.polyerrors import NotReversible
+
+# Rings
+Ring_ZZ: list[type[PowerSeriesRing]] = [PythonPowerSeriesRingZZ]
+Ring_QQ: list[type[PowerSeriesRing]] = [PythonPowerSeriesRingQQ]
+
+if GROUND_TYPES == "flint":
+    from sympy.polys.series.flint_powerseriesring import (
+        FlintPowerSeriesRingZZ,
+        FlintPowerSeriesRingQQ,
+    )
+
+    Ring_ZZ.append(FlintPowerSeriesRingZZ)
+    Ring_QQ.append(FlintPowerSeriesRingQQ)
+
+
+def same(R, s, coeffs_l, prec):
+    """
+    Helper function to assert equality of two power series.
+    """
+    domain = R.domain
+    domain_coeffs = []
+    for coeff in coeffs_l:
+        if isinstance(coeff, tuple):
+            domain_coeffs.append(domain(*coeff))
+        else:
+            domain_coeffs.append(domain(coeff))
+
+    return R.equal_repr(s, R.from_list(domain_coeffs, prec))
+
+
+@pytest.fixture(params=Ring_ZZ + Ring_QQ)
+def rd_int(request):
+    r = request.param
+    R = r()
+    x = R.gen
+    one = R.one
+
+    R3 = r(3)
+    x3 = R3.gen
+    one3 = R3.one
+
+    R10 = r(10)
+    x10 = R10.gen
+    one10 = R10.one
+    return R, x, one, R3, x3, one3, R10, x10, one10, r
+
+
+@pytest.fixture(params=Ring_QQ)
+def rd_rational(request):
+    r = request.param
+    R = r()
+    x = R.gen
+    one = R.one
+
+    R3 = r(3)
+    x3 = R3.gen
+    one3 = R3.one
+
+    R10 = r(10)
+    x10 = R10.gen
+    one10 = R10.one
+    return R, x, one, R3, x3, one3, R10, x10, one10, r
+
+
+def test_equal(rd_int):
+    R, *_ = rd_int
+    assert R.equal(R.from_list([1, 2, 3]), R.from_list([1, 2, 3])) is True
```

Comment:
I don't think it is necessary for `rd_int` to return anything more than just `r` and it would be good to give that a clearer name like `SeriesRingZZ` or something:
```
R = SeriesRingZZ()
assert R3.one ...
```